### PR TITLE
refresh-token revert and User model fix

### LINKED ISSUE #101: bug: update user stored unhashed password
**When updating the User, password isn't hashing**

- [x] update User Model to require password hashing on update request
- [x] update user test in automated tests

### LINKED ISSUE #102: Backend: Refactor verify token in refresh token
**Revert verify-token functionality into refresh-token**
- [x] make verify-token endpoint to refresh-token
- [x] refresh-token should only refresh token if the refreshToken is valid
- [x] change the details in README

### PR DESCRIPTION
**Pull request by tomdu3**
Closes #101
Closes #102 
- Refactored verify-token into refresh-token endpoint
- eliminated verification of the token (access token)
- fixed bug in User table that saved unhashed updated password

### CODE REVIEW

File: backend/controllers/authController.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-
-// Verify Token contoller - refresh if (access) token is invalid
-const verifyToken = async (req, res, next) => {
+// generate token function from .env settings
+const generateToken = (payload) => {
+  return jwt.sign(payload, process.env.JWT_SECRET, {
+    expiresIn: process.env.JWT_EXPIRES_IN,
+  });
+};
+
```
Comment (ngodi) [inline backend/controllers/authController.js:11]: I would prefer to keep jwt, and bcrypt implementations out of the authController, in a place like /lib

File: backend/controllers/authController.js [hunk 2]
```diff
@@ -22,129 +20,106 @@ const verifyToken = async (req, res, next) => {
...
+    });
+  }
 };
 
-// Signup controller
+// signup controller
 const signup = async (req, res, next) => {
-  const { username, email, password, confirmPassword, firstName, lastName, contact } = req.body;
+  const body = req.body;
+  console.log(body);
```
Comment (ngodi) [inline backend/controllers/authController.js:64]:
```suggestion

```
best practice to remove console statements when pushing code

File: backend/controllers/authController.js [hunk 3]
```diff
@@ -22,129 +20,106 @@ const verifyToken = async (req, res, next) => {
...
+        status: "fail",
+        message: "Failed to create user",
       });
+    }
+
+    return res.status(201).json({
+      status: "success",
+      message: "User created successfully",
+      data: result,
+    });
```
Comment (ngodi) [inline backend/controllers/authController.js:101]:
```suggestion
    return res.status(201).json({
      success: true/false,
      message: "User created successfully",
      data: result,
    });
```
we already have status as 201, which is a standard for created resources
success could be a boolean

**Review by ngodi — APPROVED** [review]
It's good you added tests
I'll prefer a separate folder for tests

File: backend/controllers/authController.js (hunk 3 above)
Comment (tomdu3) [reply backend/controllers/authController.js:101]: I would leave status, to follow standard messaging in the whole project

**Comment by tomdu3** [comment]
> It's good you added tests I'll prefer a separate folder for tests

It's an old discussion. My preference is to have them together with the code I'm testing.

File: backend/controllers/authController.js (hunk 2 above)
Comment (tomdu3) [reply backend/controllers/authController.js:64]: good catch

File: backend/controllers/authController.js (hunk 1 above)
Comment (tomdu3) [reply backend/controllers/authController.js:11]: I've put it in the helpers folder

**Pull request merged**